Suggest `x build library` for a custom toolchain that fails to load `core`

### Code

```Rust
x build std --target x86_64-unknown-linux-gnu
cd ../example
cargo +stage1 build --target i686-unknown-linux-gnu
```


### Current output

```Shell
error[E0463]: can't find crate for `core`
  = help: consider downloading the target with `rustup target add i686-unknown-linux-gnu`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
```


### Desired output

```Shell
error[E0463]: can't find crate for `core`
  = help: consider adding the standard library to the sysroot with `x build library --target i686-unknown-linux-gnu`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
```


### Rationale and extra context

`rustup` doesn't manage targets and components for custom toolchains. We can detect in rustc that this is a custom toolchain by checking `env!("CFG_RELEASE_CHANNEL") == "dev"` and changing the `help` output if so.

### Other cases

_No response_

### Anything else?

cc https://discord.com/channels/442252698964721669/1124482842051747860

--- a/compiler/rustc_metadata/messages.ftl
+++ b/compiler/rustc_metadata/messages.ftl
@@ -25,6 +25,9 @@ metadata_conflicting_alloc_error_handler =
 metadata_conflicting_global_alloc =
     the `#[global_allocator]` in {$other_crate_name} conflicts with global allocator in: {$crate_name}
 
+metadata_consider_adding_std =
+    consider adding the standard library to the sysroot with `x build library --target {$locator_triple}`
+
 metadata_consider_building_std =
     consider building the standard library from source with `cargo build -Zbuild-std`
 

--- a/compiler/rustc_metadata/src/errors.rs
+++ b/compiler/rustc_metadata/src/errors.rs
@@ -646,12 +646,18 @@ impl IntoDiagnostic<'_> for CannotFindCrate {
             } else {
                 diag.note(fluent::metadata_target_no_std_support);
             }
-            // NOTE: this suggests using rustup, even though the user may not have it installed.
-            // That's because they could choose to install it; or this may give them a hint which
-            // target they need to install from their distro.
+
             if self.missing_core {
-                diag.help(fluent::metadata_consider_downloading_target);
+                if env!("CFG_RELEASE_CHANNEL") == "dev" {
+                    diag.help(fluent::metadata_consider_adding_std);
+                } else {
+                    // NOTE: this suggests using rustup, even though the user may not have it installed.
+                    // That's because they could choose to install it; or this may give them a hint which
+                    // target they need to install from their distro.
+                    diag.help(fluent::metadata_consider_downloading_target);
+                }
             }
+
             // Suggest using #![no_std]. #[no_core] is unstable and not really supported anyway.
             // NOTE: this is a dummy span if `extern crate std` was injected by the compiler.
             // If it's not a dummy, that means someone added `extern crate std` explicitly and

